sources/packages/web/src/components/common/sin/AddNewSIN.vue

```diff
@@ -0,0 +1,143 @@
+          hide-details
+          label="Social Insurance Number (SIN)"
+          v-model="formModel.sin"
+          variant="outlined"
+          :rules="[(v) => sinValidation(v)]" />
+        <v-checkbox
+          label="Skip the validations"
+          v-model="formModel.skipValidations"
+          hide-details />
+        <banner
+          class="mb-4"
+          v-if="formModel.skipValidations"
+          :type="BannerTypes.Error"
+          header="Danger! Are you sure you want to skip the validations?"
+          summary=" By selecting this, you are forcing the SIN to be valid for the
+            student. Please proceed with caution."
+        >
+        </banner>
+        <v-textarea
+          hide-details
+          label="Notes"
+          placeholder="Long text..."
+          v-model="formModel.noteDescription"
+          variant="outlined"
+          :rules="[(v) => !!v || 'Notes is required']"
+      /></template>
+      <template #footer>
+        <check-permission-role :role="Role.StudentAddNewSIN">
+          <template #="{ notAllowed }">
+            <footer-buttons
+              justify="end"
+              primaryLabel="Add SIN now"
+              @secondaryClick="cancel"
+              @primaryClick="submit"
+              :disablePrimaryButton="notAllowed"
+            />
+          </template>
+        </check-permission-role>
+      </template>
+    </modal-dialog-base>
+  </v-form>
+</template>
+<script lang="ts">
+import { PropType, ref, reactive } from "vue";
+import ModalDialogBase from "@/components/generic/ModalDialogBase.vue";
+import ErrorSummary from "@/components/generic/ErrorSummary.vue";
+import { useModalDialog } from "@/composables";
+import { Role, VForm } from "@/types";
+import CheckPermissionRole from "@/components/generic/CheckPermissionRole.vue";
+import { CreateSINValidationAPIInDTO } from "@/services/http/dto";
+import { BannerTypes } from "@/types/contracts/Banner";
+
+export default {
+  components: { ModalDialogBase, CheckPermissionRole, ErrorSummary },
+  props: {
+    entityType: {
+      type: String,
+      required: true,
+    },
+    allowedRole: {
+      type: String as PropType<Role>,
+      required: true,
+    },
+  },
+  setup() {
+    const { showDialog, showModal, resolvePromise } = useModalDialog<
+      CreateSINValidationAPIInDTO | boolean
+    >();
+    const addNewSINForm = ref({} as VForm);
+    const formModel = reactive({} as CreateSINValidationAPIInDTO);
+
+    const submit = async () => {
+      const validationResult = await addNewSINForm.value.validate();
+      if (!validationResult.valid) {
+        return;
+      }
+      const payload = { ...formModel };
+      resolvePromise(payload);
+      addNewSINForm.value.reset();
+    };
+
+    const cancel = () => {
+      addNewSINForm.value.reset();
+      formModel.skipValidations = false;
+      addNewSINForm.value.resetValidation();
+      resolvePromise(false);
+    };
+
+    const sinValidation = (sin: string) => {
```

Comment:
We are doing SIN validation in the student search also, there is already a method present on Vue. Can we please reuse that?